src/defs.hpp

```diff
@@ -137,6 +138,19 @@ using MGBoundaryFunc = void (*)(ParArrayND<Real> &dst, Real time, int nvar, int
                                 int ie, int js, int je, int ks, int ke, int ngh, Real x0,
                                 Real y0, Real z0, Real dx, Real dy, Real dz);
 
+//----------------------------------------------------------------------------------------
+// Opaque pointer to application data
+class MeshBlockApplicationData {
```

Comment:
I don't know what this is, I just saw that it was used by a function that could be weakly linked. I had to move it out of `mesh.hpp` to avoid a circular dependency.

Reply:
Oh my bad. I thought you wrote that.